### ISSUE TITLE
LJPME is implemented but not a part of the SMIRNOFF spec

### ISSUE THREAD

**Issue by mattwthompson** [openforcefield/openff-toolkit#989]
**Is your feature request related to a problem? Please describe.**
LJPME is implemented in some capacity in the toolkit, but it not part of the [SMIRNOFF spec](https://open-forcefield-toolkit.readthedocs.io/en/0.9.1/smirnoff.html#vdw). This adds a decent amount of complexity to handling and testing non-bonded methods/OpenMM exports. 

* Allowed as a `vdWHandler.method` value: https://github.com/openforcefield/openff-toolkit/blob/1ba11ebbafa67f7953b13f6c0af6183a5801f33e/openff/toolkit/typing/engines/smirnoff/parameters.py#L3575-L3577
* Somewhat dealt with when wrangling OpenMM forces: https://github.com/openforcefield/openff-toolkit/blob/1ba11ebbafa67f7953b13f6c0af6183a5801f33e/openff/toolkit/typing/engines/smirnoff/parameters.py#L3638-L3649
* More dealing with `NonbondedForce` quirks: https://github.com/openforcefield/openff-toolkit/blob/1ba11ebbafa67f7953b13f6c0af6183a5801f33e/openff/toolkit/typing/engines/smirnoff/parameters.py#L3876-L3885
* Added test complexity: https://github.com/openforcefield/openff-toolkit/blob/master/openff/toolkit/tests/test_forcefield.py#L363-L460

It's [covered](https://codecov.io/gh/openforcefield/openff-toolkit/src/master/openff/toolkit/typing/engines/smirnoff/parameters.py) in CodeCov's opinion, but I don't think any substantive tests cover LJPME.

**Describe the solution you'd like**
It seems straightforward to suggest that either the implementation should be removed or it should be added to the SMIRNOFF spec.

**Describe alternatives you've considered**
We could leave it be, but that's likely to produce maintenance issues, bad behavior and/or user experience; implementation should follow specification, not vice versa. Deprecating and then removing LJPME functionality should be straightforward, or expanding the SMIRNOFF spec to allow `method="pme"` in the vdW section (https://github.com/openforcefield/standards/issues/11) would sync things up.

**Additional context**
As far as I know, nobody is using LJPME from the toolkit; if they were, it might take a significant amount of modifications (either tinkering with the OpenMM System or using a force field that does not follow SMIRNOFF).

I don't think re-fitting with LJPME is on the next year or so of the science roadmap, but there might be some interest in using it long-term. I'm pretty sure it's implemented in all major simulation engines.

I'm probably missing some history, but I [couldn't quickly](https://github.com/openforcefield/openff-toolkit/search?q=ljpme&type=issues) find anything documented on this.

**Comment by jchodera** [comment]
I would recommend that LJPME be added to the spec. I think this was originally our intention, but it must have slipped through the cracks.

There is now significant evidence that LJPME is superior to the standard PME + LJ-cutoff/analytical dispersion correction approach in multiple applications of import:
* [ligand binding free energies](https://pubs.acs.org/doi/abs/10.1021/jp0735987)
* [lipid bilayers](https://pubs.acs.org/doi/abs/10.1021/ct400140n)
* [organic liquids](https://pubs.acs.org/doi/abs/10.1021/acs.jctc.7b00948)

Since [recent algorithmic improvements](https://pubs.acs.org/doi/abs/10.1021/acs.jctc.5b00726) have made it possible to carry out LJPME simulations even for Lorentz-Berthelot combining rules, and LJPME is supported by [OpenMM](http://docs.openmm.org/latest/userguide/application.html), [gromacs](https://manual.gromacs.org/documentation/2019/reference-manual/functions/long-range-vdw.html#using-lj-pme), [CHARMM](https://pubs.acs.org/doi/abs/10.1021/acs.jctc.0c01327), and [AMBER](https://ambermd.org/doc12/Amber20.pdf), there is no risk in parameterizing a force field with it.

Just as we should be considering producing reaction field and standard PME variants of our main line force fields, we should seriously consider producing an LJPME variant with improved properties.

**Comment by mattwthompson** [comment]
After #1679, "PME" is no longer an allowed value for `vdWHandler.method` (or its relevant replacement, `vdWHandler.periodic_method`).

This should be interpreted as backing away from LJPME - I believe we're all in agreement that we want it in our infrastructure and would like to at least explore its use in mainline OpenFF force fields - but the toolkit should avoid filling in gaps in the SMIRNOFF spec. Once some form of LJPME is in the spec, I hope we can add it in the toolkit and Interchange in short time.

**Comment by jchodera** [comment]
We should not "back away from LJPME" since it is a best practice.

See this proposal to amend the SMIRNOFF spec to address the LJPME issue from July 2022:
https://github.com/openforcefield/standards/pull/44

**Comment by mattwthompson** [comment]
I have a knack for unhelpful typos and this is one such instance - I mean to say this should **NOT** be interpreted as backing away from LJPME! My mistake, sorry for the confusion!
